pkg/cluster/keyvaults.go

```diff
@@ -0,0 +1,77 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+	"github.com/Azure/ARO-RP/pkg/util/uuid"
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2019-09-01/keyvault"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (m *manager) createKeyvault(ctx context.Context) error {
+	// We want random characters from end of infraID to help ensure KV name uniqueness
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+	suffixMaxLength := 20
+	var suffix string
+
+	// This runs after ensureInfraID bootstrap step, but just in case
+	if infraID == "" {
+		suffix = m.doc.OpenShiftCluster.Name
+	} else {
+		suffix = infraID
+	}
+
+	suffixLength := len(suffix)
+	if suffixLength > suffixMaxLength {
+		suffix = suffix[suffixLength-suffixMaxLength : suffixLength]
```

Comment:
what are we trying to achieve here? couldn't we just truncate the suffix like this? 
```go
suffix = suffix[:suffixMaxLength]
```

(also you can write `suffix = suffix[suffixLength-suffixMaxLength:]`  which would be equivalent to what you have right now, it simplifies it)

Reply:
The comment from line 17 still applies here, we want the _end_ of the `InfraID` that has the randomly-generated characters because KV names need to be globally unique. I could just grab the 5 characters at the end that we randomly generate, there are 36^5 > 60 million possible combinations, so the likelihood of collision is small, but including as much of the infraID with the cluster name as possible reduces that likelihood even further.